gplugins/design_recipe/DesignRecipe.py

```diff
@@ -25,57 +27,64 @@ class DesignRecipe:
     # the hash of the system last time eval() was executed
     last_hash: int
 
-    # The component this DesignRecipe operates on. This is not necessarily
-    # the same `component` refered to in the `dependencies` recipes.
-    component: Optional[Component] = None
+    # The component factory this DesignRecipe operates on. This is not necessarily
+    # the same `component` referred to in the `dependencies` recipes.
+    cell: ComponentFactory | Component | None = None
```

Comment:
`ComponentFactory` is a callable, right? Would `cell` need to either be evaluated or not evaluated (to get the design) depending on if it's a `ComponentFactory` or a `Component`?

Reply:
Ah I see that logic is handled- curious, why might we want to have `cell` be a `ComponentFactory`? Couldn't we cast it to a `Component` before passing it into a DR?

Reply:
What do you mean by casting? Are you still able to perform a geometry sweep? If not, a ComponentFactory is more applicable since we can sweep its geometry parameters